### ISSUE TITLE
Add the ability to list connected cavities

### ISSUE THREAD

**Issue by jmaglic** [jmaglic/MoloVol#71]
### Issue

We have observed structures in which, due to the choice of parameters, cavities that should be connected appear fractured. One such example is:
```
probetest_quadruplet.xyz
r1 = 0.5
r2 = 5.0
g = 0.1
d = 4
```
This appears to be an inevitable issue with the discrete nature of voxels. However, a user may be able to determine for themselves whether two cavities should be treated as one, or not. This may require viewing the surface maps in an external program.

### Proposed change

I suggest a feature that allows finding all neighbouring cavities of a selected cavity. After selecting a cavity, a list of cavity IDs can be provided.

### Improvement

This feature would make it much easier to find connected cavities, especially in larger molecules, where the number of cavities may be 10+. This would be particularly useful for evaluating very small cavities, since it would allow quickly selecting which surface maps to view.

**Comment by rlavendomme** [comment]
Such a feature would make it more convenient for users to understand the relation between cavities but it not critical as they can already do it. I propose to add this feature after the release.

A simple solution would be to suggest to use a finer grid in case the number of cavity is higher than expected by the user. For the above example, the cavity is not fractured anymore with a gridstep of 0.07 Å or smaller.

> Proposed change
> 
> I suggest a feature that allows finding all neighbouring cavities of a selected cavity. After selecting a cavity, a list of cavity IDs can be provided.

We would need to decide arbitrarily what is a neighboring cavity. Would "cavities separated by less than twice the gridstep" make sense ?

**Comment by jmaglic** [comment]
Agreed that this is not needed right now. I'm opening the issue only for us to keep the feature in mind for later.

> We would need to decide arbitrarily what is a neighboring cavity. Would "cavities separated by less than twice the gridstep" make sense ?

As neighbour I would only consider those cavities that are "touching", i.e. there are two (shell) voxels of different IDs right next to each other. I feel like there's no arbitrariness.

> A simple solution would be to suggest to use a finer grid in case the number of cavity is higher than expected by the user. For the above example, the cavity is not fractured anymore with a gridstep of 0.07 Å or smaller.

This should definitely be added to the user manual.
